File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int i = 0;
        int j = numbers.size()-1;

        while (i < j) {
            int sum = numbers[i] + numbers[j];
            if (sum == target) {
                break;
            } else if (sum > target) {
                j--;
            } else {
                i++;
            }
        }

        vector<int> ans;
        ans.push_back(i+1);
        ans.push_back(j+1);

        return ans;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        if len(nums) == 2:
            return [0, 1]
        
        seen = {}

        for i in range(len(nums)):
            pair = target - nums[i]
            if pair in seen.keys():
                return [seen[pair], i]
            
            if nums[i] not in seen.keys():
                seen[nums[i]] = i

        return [0, 1]

        